packages/hardhat-ethers/src/internal/ethers-websocket-provider-proxy.ts

```diff
@@ -0,0 +1,36 @@
+import { ethers } from "ethers";
+
+import {
+  HARDHAT_NETWORK_RESET_EVENT,
+  HARDHAT_NETWORK_REVERT_SNAPSHOT_EVENT,
+} from "hardhat/internal/constants";
+import { EthereumProvider } from "hardhat/types";
+
+import { createUpdatableTargetProxy } from "./updatable-target-proxy";
+
+/**
+ * This method returns a proxy that uses an underlying provider for everything.
+ *
+ * This underlying provider is replaced by a new one after a successful hardhat_reset,
+ * because ethers providers can have internal state that returns wrong results after
+ * the network is reset.
+ */
+export function createProviderProxy(
```

Comment:
Is this only meant for testing?

Reply:
Sorry, I should have been more clear here.  This change was only made in support of the websocket+ganache test.  To be honest I hacked my way through that test to make it work.  Maybe it's actually not as ready for "merge" as the rest of the tests in here.

Reply:
Eliminated this in https://github.com/nomiclabs/hardhat/pull/1792/commits/3ec7a887d380aaaa1e7e119ad15dcc96d94d3a82